code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -100,19 +94,25 @@
 	if (mode.check_lowpop_lowimpact_injection())
 		return FALSE
 
+	if (!..()) // calls check_candidates()
+		return FALSE
+
 	return TRUE
 
-/datum/dynamic_ruleset/midround/from_ghosts/execute()
+/datum/dynamic_ruleset/midround/from_ghosts/execute(forced = FALSE)
 	var/list/possible_candidates = list()
 	possible_candidates.Add(dead_players)
 	possible_candidates.Add(list_observers)
-	send_applications(possible_candidates)
-	return length(assigned)
+	send_applications(possible_candidates, forced)
+	return length(assigned) ? TRUE : NOT_ENOUGH_PLAYERS
 
 /// This sends a poll to ghosts if they want to be a ghost spawn from a ruleset.
-/datum/dynamic_ruleset/midround/from_ghosts/proc/send_applications(list/possible_volunteers = list())
+/datum/dynamic_ruleset/midround/from_ghosts/proc/send_applications(list/possible_volunteers = list(), forced = FALSE)
 	if (!length(possible_volunteers)) // This shouldn't happen, as ready() should return FALSE if there is not a single valid candidate
-		message_admins("Possible volunteers was 0. This shouldn't appear, because of ready(), unless you forced it!")
+		if(!forced)
+			message_admins("Possible volunteers was 0. This shouldn't appear, because of ready()!")
+			log_game("DYNAMIC: Possible volunteers was 0. This shouldn't appear, because of ready()!")
+			CRASH("The ruleset [name] execute()d with no candidates. This should have been caught by ready(), so something is wrong.")
```

Comment:
Pretty sure crash already gets logged

Reply:
Having it in both logs makes it much easier to see and debug